### DISCUSSION TITLE
Do not retrieve articles that are for paid subscribers from substack

### DISCUSSION THREAD

**Discussion by riverofjanuary** [FreshRSS/FreshRSS#6667] [Q&A] [answered]
Hi all

I use Freshrss to get a lot of substacks, and it works great. But about half my substack have free and paid entries. Can I filter out the paid ones and not even retrieve them somehow?  

Let's use this as example of 'paid' : https://adamtooze.substack.com/p/top-links-477-the-irreversible-covid

There is no 'paid' class or anything on the body. But there is a div classed 'paywall-title' which contains "This post is for paid subscribers". Can I block an entry (or maybe mark it read/not retrieve it) based on this somewhere?

**Comment by Alkarex** [comment]
If there is some text in the title or body, which can help discriminating which articles are free or not, then you can automatically mark as read the articles you do not want to see

**Reply by riverofjanuary** [reply]
please see my reply below. sometimes that text on its own is insufficient to filter and can cause false positives. thanks.

**Comment by riverofjanuary** [comment]
Just wanted to add a note that not all substacks use the same language ("for paid subscribers") and some use that language, but not on a paid post (eg "For paid subscribers, we have more things coming up next weekend!").  So the div "paywall-title" is likely the best, or a combination of text + class?

**Reply by Alkarex** [reply]
You need to experiment a bit. You can use multiple rules and combine positive and negative assertions
https://freshrss.github.io/FreshRSS/en/users/10_filter.html#with-the-search-field

**Reply by riverofjanuary** [reply]
Is there a way to filter by a div class or ID? eg ':has:div.paywall-title'?

**Reply by Alkarex** [reply]
Not at the moment, but you can try by plaintext matching

**Reply by prank-arum** [reply]
Bumping this on behalf of the OP: plaintext matching does not find the text within the div classed 'paywall-title' on Substack, even though it is visible and does show up when doing a regular search in, say Firefox. So as of now there doesn't seem to be a way to parse these articles.

**Comment by upstreamist** [comment, marked as answer]
Filtering the phrase ``Read more`` should do it. That seems to be the key phrase Substack automatically appends to RSS items that have been truncated because the rest of the content is paywalled. That may also occasionally catch posts that contain the phrase "read more" elsewhere in the body text, though. Once 1.25.0+ ships, the regex filter ``/^Read more$/m`` should work better.

**Reply by riverofjanuary** [reply]
> That may also occasionally catch posts that contain the phrase "read more" elsewhere in the body text, though

This was my concern up top. Many (free) posts use those words elsewhere ("Read more about the elections in this person's article") or similar.

How would the regex prevent this?

**Reply by upstreamist** [reply]
The ^ and $ symbols indicate start and end of line, so ``/^Read more$/m`` should catch *only* posts that have that exact phrase one a line by themself, without filtering posts that have the phrase embedded in a sentence.

**Reply by Alkarex** [reply]
It looks like there might be some whitespace before, so maybe something like `/^\s*Read more\s*$/m`

**Reply by riverofjanuary** [reply]
Ok friends, thank you. Looks like 1.25.0+ release is far-ish off, unfortunately, so I'll make due til then.

**Reply by Alkarex** [reply]
You are welcome to use our rolling release. No need to wait for the next slow release

**Reply by riverofjanuary** [reply]
I might try that, thanks.

**Comment by prank-arum** [comment]
Hi all, I came up with a different (I think, better) way to do this. I don’t use the Substack RSS feed anymore. Instead, I scrape the site and use some logic to set an article tag based upon whether the padlock icon is present. This has worked really well.
I’m happy to share the code when I get home, if anyone wants it.

**Reply by riverofjanuary** [reply]
Very interesting! Would love to see the code.

Did you unsubscribe and re-add all the substacks you had before after you figured this out? That would be onerous but I guess doable for an hour or two!

**Comment by prank-arum** [comment]
I did have to re-add everything, but I don't follow that many Substack writers so it wasn't a big deal.
Here's a link to a blog post I made about it. Let me know how it works out:

[https://djlcomputertips.blogspot.com/2024/10/mark-paid-substack-posts-as-read-in.html](https://djlcomputertips.blogspot.com/2024/10/mark-paid-substack-posts-as-read-in.html)

**Reply by riverofjanuary** [reply]
Thanks! Now just have to figure out the best way to install 1.25....

**Reply by Alkarex** [reply]
> Now just have to figure out the best way to install 1.25....

You can use our rolling release `edge`. Available by Docker, git, ZIP.
